lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -166,9 +166,18 @@ public int getMegaBytes() {
     private Builder() {
     }
 
+    // setting builder to the values passed in the request
     private Builder(final EvaluatorRequest request) {
       setNumber(request.getNumber());
       fromDescriptor(request.getDescriptor());
+      setMemory(request.getMegaBytes());
```

Comment:
Kind of. I added this functionality because I need it now, makes life easier on the dataLoader. Apparently before nobody needed it. 
As you wish, I can leave it like this for now, or do all the process and create a new pull request for this, in a separate branch and so on.
I think that as the change is so small, it may not be worth investing time in creating a separate jira. But is your call.